### ISSUE TITLE
Support fetch TLS client certificate

### ISSUE THREAD

**Issue by iugo** [denoland/deno#10516]
Just like `curl --cert <file.pem>`.

Now the way to solve this problem in deno is: curl. But this method is not cross-platform.

**Comment by DjDeveloperr** [comment]
I think https://doc.deno.land/builtin/unstable#Deno.createHttpClient works for that.

**Comment by lucacasonato** [comment]
We already support this with createHttpClient and the --cert flag.

**Comment by danopia** [comment]
Actually no? Curl's `--cert` is for _client certificates_ and Deno's `--cert` is for _certificate authorities_ so the naming is pretty confusing. But afaik Deno does not offer what this issue asks for: TLS client certificates in `fetch()`.

deno:
```
        --cert <FILE>                  
            Load certificate authority from PEM encoded file
```

curl:
```
     --cacert <file> CA certificate to verify peer against

 -E, --cert <certificate[:password]> Client certificate file and password
     --cert-type <type> Certificate file type (DER/PEM/ENG)
```

**Comment by lucacasonato** [comment]
You're right. Reopened.

**Comment by justinmchase** [comment]
Proposal:
```ts
interface CreateHttpClientOptions {
  // A certificate authority to use when validating TLS certificates. Certificate data must be PEM encoded.
  ca?: string
  // Client certificate chains in PEM format. 
  cert?: string
  // Client certificate private keys in PEM format.
  key?: string

  proxy?: Proxy
}
```

These names are mimicking the nodejs api for tls options:
https://nodejs.org/docs/latest-v15.x/api/tls.html#tls_tls_createsecurecontext_options

You can see there are a ton of other options but most of the time if you're using client certificates you just need the `cert` and `key` fields. This will help a lot of usecases without having to go nuts implementing every option under the sun.

**Comment by justinmchase** [comment]
Related to https://github.com/denoland/deno/issues/9121

**Comment by wesleyakio** [comment]
Hey guys, I spent a few hours reading through the issues and it seems like this landed on main about two years ago in #11721 but I couldn't find anything definitive as to why this is still behind `--unstable`. Cursory tests show that this works for my purposes but I'd love to also use this on Deno Deploy so understanding what kind of timeframe are we working with for this to land on `stable` would be very helpful. Anyone around could shed some light?
